plugins/woocommerce/includes/class-wc-comments.php

```
@@ -306,6 +317,67 @@ public static function delete_comments_count_cache() {
 		wp_cache_delete_multiple( $comment_status_keys, self::COMMENT_COUNT_CACHE_GROUP );
 	}
 
+	/**
+	 * Fetches (and populates if needed) the counter.
+	 *
+	 * @return int
+	 */
+	public static function get_products_reviews_pending_moderation_counter(): int {
+		$count = wp_cache_get( self::PRODUCT_REVIEWS_PENDING_COUNT_CACHE_KEY, self::COMMENT_COUNT_CACHE_GROUP );
+		if ( false === $count ) {
+			$count = (int) get_comments(
+				array(
+					'type__in'  => array( 'review', 'comment' ),
+					'status'    => '0',
+					'post_type' => 'product',
+					'count'     => true,
+				)
+			);
+			wp_cache_set( self::PRODUCT_REVIEWS_PENDING_COUNT_CACHE_KEY, $count, self::COMMENT_COUNT_CACHE_GROUP, DAY_IN_SECONDS );
+		}
+
+		return $count;
+	}
+
+	/**
+	 * Handles `wp_insert_comment` hook processing and actualizes the counter.
+	 *
+	 * @param int         $comment_id Comment ID.
+	 * @param \WP_Comment $comment    Comment object.
+	 * @return void
+	 */
+	public static function maybe_bump_products_reviews_pending_moderation_counter( $comment_id, $comment ): void {
+		$needs_bump = '0' === $comment->comment_approved;
+		if ( $needs_bump && in_array( $comment->comment_type, array( 'review', 'comment', '' ), true ) ) {
+			$is_product = 'product' === get_post_type( $comment->comment_post_ID );
+			if ( $is_product && false !== wp_cache_get( self::PRODUCT_REVIEWS_PENDING_COUNT_CACHE_KEY, self::COMMENT_COUNT_CACHE_GROUP ) ) {
```

I think you can skip the `false !== wp_cache_get()` check here.  The incr/decr functions should just silently fail if the values don't exist.

---

Sounds good: simplified in c3f882c3ad09a038849fbbd233fbcbd25654ab79!